### ISSUE TITLE
travis: tests hang in valgrind

### ISSUE THREAD

**Issue by chu11** [flux-framework/flux-core#1311]
Atleast 4 times, I've seen travis hang after the 

``` PASS: t4000-issues-test-driver.t 4 - t0900-wreck-invalid-cores.sh```

test.  (#1301, #1286, #1265, #1299).  The test after this is valgrind.  What could cause the valgrind test to start, but never finish?

**Comment by grondo** [comment]
I've never seen the test hang, but it is definitely slow. I wonder if the valgrind test could slow the code enough to tickle a hang that is not otherwise seen?

**Comment by chu11** [comment]
My feeling is its probably just slow, but was wondering why it didn't appear before.  Then I recalled #1245 removing "-j2".  Wondering if we just were able to get by this b/c the tests were running in parallel before.

**Comment by grondo** [comment]
yeah, but the travis timeout indicates no output for 10m, which means it is just the valgrind test that is running this long, which is pretty unusual (assuming it is actually the valgrind test and not something else that is stuck)

One thing I've tried in the past to diagnose this kind of thing is adding a line to travis to run the test by hand in the foreground to see if I can trigger the issue. The `make check` run sharness tests get their output eaten up by the tap-driver.sh, so you can't see how far the test gets... There may also be some data in the test log dropped into the ./t dir
